datalad/distribution/dataset.py

```diff
@@ -321,6 +349,8 @@ def repo(self):
             # at DEBUG level and if necessary "complaint upstairs"
             lgr.log(5, "Failed to detect a valid repo at %s", self.path)
         elif due.active:
+            # TODO: Figure out, when exactly this is needed. Don't think it makes sense to do this for every dataset,
+            #       no matter what => we want .repo to be as cheap as it gets.
             # Makes sense only on installed dataset - @never_fail'ed
             duecredit_dataset(self)
```

Comment:
This seems to be expensive indeed:

```
>>> ds=dl.Dataset('.')
>>> timeit duecredit_dataset(ds)                                                                                         
[WARNING] Found no aggregated metadata info file /home/mih/hacking/datalad/git/.datalad/metadata/aggregate_v1.json. You will likely need to either update the dataset from its original location or reaggregate metadata locally. 
...
98.7 ms ± 3.35 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# for a dataset with actual metadata
>>> ds=dl.Dataset('/tmp/studyforrest-data-phase2/')
timeit duecredit_dataset(ds)                                                                                         
>>> 141 ms ± 8.48 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

However, I think this is only a concern when running "inside" duecredit and has no effect under normal operations. Is that correct @yarikoptic ?